src/Handler/Table/Import/ImportTableFromTableHandler.php

```diff
@@ -219,32 +220,52 @@ private function import(
             return [null, $importState->getResult()];
         }
 
-        $stagingTable = $this->createStageTable($destinationDefinition, $sourceMapping, $bqClient);
+        /** @var TableImportFromTableCommand\SourceTableMapping\ColumnMapping[] $mappings */
+        $mappings = iterator_to_array($sourceMapping->getColumnMappings()->getIterator());
+        $stagingTable = StageTableDefinitionFactory::createStagingTableDefinitionWithMapping(
+            $destinationDefinition,
+            $mappings,
+        );
```

Comment:
To vytváření staging tabulky bych přesunul do `if ($isColumnNameIdentical) {` k `CopyImportFromTableToTable`

Reply:
Chvilku sem nad tím přemýšlal a změnil sem to, dělala sa tam ta stage tabulka 2x stejně